nixos/modules/services/web-apps/moin.nix

```diff
@@ -0,0 +1,273 @@
+          };
+
+          extraConfig = mkOption {
+            type = lines;
+            default = "";
+            example = ''
+              search_results_per_page = 100
+              acl_rights_default = u"Known:read,write,delete,revert All:read"
+              logo_string = u"<h2>\U0001f639</h2>"
+              theme_default = u"modernized"
+
+              user_checkbox_defaults = {'show_page_trail': 0, 'edit_on_doubleclick': 0}
+              navi_bar = [u'SomePage'] + multiconfig.DefaultConfig.navi_bar
+              actions_excluded = multiconfig.DefaultConfig.actions_excluded + ['newaccount']
+            '';
+            description = ''
+              Additional configuration to be appended verbatim to this wiki's config.
+
+              See <link xlink:href='http://moinmo.in/HelpOnConfiguration' /> for documentation.
+            '';
+          };
+
+        };
+        config = {
+          webHost = mkDefault name;
+        };
+      }));
+      example = literalExample ''
+        {
+          "mywiki" = {
+            siteName = "Example Wiki";
+            webHost = "wiki.example.org";
+            superUsers = [ "admin" ];
+            frontPage = "Index";
+            extraConfig = "page_category_regex = ur'(?P<all>(Category|Kategorie)(?P<key>(?!Template)\S+))'"
+          };
+        }
+      '';
+      description = ''
+        Configurations of the individual wikis. Keys must be valid Python
+        identifiers of the form <literal>[A-Za-z_][A-Za-z0-9_]*</literal>.
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = lines;
+      default = "";
+      example = ''
+        mail_smarthost = "mail.example.org"
+        mail_from = u"Example.Org Wiki <wiki@example.org>"
+      '';
+      description = ''
+        Additional configuration for all wikis.
+
+        See <link xlink:href='http://moinmo.in/HelpOnConfiguration' /> for documentation.
+      '';
+    };
+
+    webServer = mkOption {
+      type = enum [ "nginx-uwsgi" "none" ];
+      default = "nginx-uwsgi";
+      example = "none";
+      description = ''
+        Which web server to use to serve the wiki.
+        Use <literal>none</literal> if you want to configure this yourself.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = flip map (attrNames cfg.wikis) (wname:
+      { assertion = builtins.match "[A-Za-z_][A-Za-z0-9_]*" wname != null;
+        message = "${wname} is not valid Python identifier";
+      }
+    );
+
+    users.users = {
+      moin = {
+        description = "MoinMoin wiki";
+        home = dataDir;
+        group = group;
+        createHome = false;
+        isSystemUser = true;
+      };
+    };
+
+    users.groups = {
+      moin = {};
+      "${config.services.uwsgi.group}".members = mkIf isNginxUwsgi [ config.services.nginx.user ];
+    };
+
+    environment.systemPackages = [ moinCliWrapper ];
+
+    services.uwsgi = mkIf isNginxUwsgi {
+      enable = true;
+      plugins = [ "python2" ];
```

Comment:
*I think* that if you remove this and add `plugin = "python2"` to the moin vassal configuration, the same uwsgi instance/emperor can be used for Python 2 and Python 3. (untested)

Reply:
Removing this results in the module pulling in uwsgi package that is built without any plugins. I'll wrap the option in `mkDefault` so that adding more plugins is easier.